pandas/core/arrays/interval.py

```diff
@@ -1435,6 +1438,40 @@ def contains(self, other):
             other < self._right if self.open_right else other <= self._right
         )
 
+    def isin(self, values) -> np.ndarray:
+        if not hasattr(values, "dtype"):
+            values = np.array(values)
+        values = extract_array(values, extract_numpy=True)
+
+        if is_interval_dtype(values.dtype):
+            if self.closed != values.closed:
+                # not comparable -> no overlap
+                return np.zeros(self.shape, dtype=bool)
+
+            if is_dtype_equal(self.dtype, values.dtype):
+                # error: "ArrayLike" has no attribute "view"  [attr-defined]
+                left = self._combined.view("complex128")  # type:ignore[attr-defined]
```

Comment:
really?

Reply:
can you be more specific?

Reply:
its the conversion to complex128. can you at least comment on why you are doing it

Reply:
ill add a comment